x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -106,6 +117,12 @@ export function getMigrations(
     pipeMigrations(setLegacyId, getRemovePreconfiguredConnectorsFromReferencesFn(isPreconfigured))
```

Comment:
Can you add the `addRuleIdsToLegacyNotificationReferences` to this migrations pipe and inside this function, make sure it filters for security only rules?

```
function addRuleIdsToLegacyNotificationReferences(
  doc: SavedObjectUnsanitizedDoc<RawAlert>
): SavedObjectUnsanitizedDoc<RawAlert> {
  if (!isSecuritySolutionLegacyNotification(doc)) {
    return doc
  }

  .. rest of function
}
```